libs/MobileSync/MobileSync/Classes/Manager/SFAdvancedSyncUpTask.m

```diff
@@ -32,72 +33,87 @@ -(instancetype) init:(SFMobileSyncSyncManager*)syncManager sync:(SFSyncState*)sy
 }
 
 - (void)syncUp:(SFSyncState*)sync recordIds:(NSArray*)recordIds {
-    [self syncUpMultipleEntries:sync recordIds:recordIds index:0 batch:[NSMutableArray new]];
+    SFSyncUpTarget *target = (SFSyncUpTarget *)sync.target;
+    NSString* soupName = sync.soupName;
+    sync.totalSize = recordIds.count;
+    
+    NSArray<NSDictionary*>* records = [target getFromLocalStore:self.syncManager
+                                                       soupName:soupName
+                                                       storeIds:recordIds];
+    
+    // Figuring out what records need to be synced up based on merge mode and last mod date on server
+    [self shouldSyncUpRecords:self.syncManager target:target records:records options:sync.options resultBlock:^(NSDictionary * _Nonnull recordIdToShouldSyncUp) {
```

Comment:
Same refactoring we did on Android.

Reply:
To allow sync up target to decide if records should be synched in bulk (but only CollectionSyncUpTarget actually does the checking in bulk). Existing sync up target check the old way. Two reasons for leaving them alone:
- more motivation for customer to adopt the new CollectionSyncUpTarget
- not break customer that might have store multiple types of objects in the sam e soup: it's not forbidden and works with BatchSyncUpTarget but is not allowed with the new CollectionSyncUpTarget.